addon/components/-private/scroll-indicators/component.js

```diff
@@ -0,0 +1,135 @@
+/* global ResizeSensor */
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { readOnly } from '@ember/object/computed';
+import { bind } from '@ember/runloop';
+import { htmlSafe } from '@ember/template';
+import { isEmpty } from '@ember/utils';
+import { addObserver } from 'ember-table/-private/utils/observer';
+import layout from './template';
+import { getScale } from 'ember-table/-private/utils/element';
+
+const indicatorStyle = side => {
+  return computed(
+    `columnTree.${side}FixedNodes.@each.width`,
+    'scrollRect',
+    'tableRect',
+    function() {
+      let style = [];
+
+      // left/right position
+      let fixedNodes = this.get(`columnTree.${side}FixedNodes`);
+      let scrollbarOffsets = { left: 0, right: this.get('scrollbarOffset') };
+      if (!isEmpty(fixedNodes)) {
+        let fixedWidth = fixedNodes.reduce((acc, node) => acc + node.get('width'), 0);
+        style.push(`${side}:${fixedWidth + scrollbarOffsets[side]}px;`);
+      } else {
+        style.push(`${side}:${scrollbarOffsets[side]}px;`);
+      }
+
+      // height
+      let scrollRect = this.get('scrollRect');
+      let tableRect = this.get('tableRect');
+      if (scrollRect && tableRect) {
+        style.push(`height:${Math.min(scrollRect.height, tableRect.height)}px;`);
+      }
+
+      return htmlSafe(style.join(''));
+    }
+  );
+};
+
+export default Component.extend({
+  layout,
+  tagName: '',
+
+  /**
+    The API object passed in by the table
+
+    @argument api
+    @required
+    @type object
+  */
+  api: null,
+
+  showLeft: true,
+  showRight: false,
+
+  columnTree: readOnly('api.columnTree'),
+  enableScrollIndicators: readOnly('api.enableScrollIndicators'),
+  tableScrollId: readOnly('api.tableId'),
+
+  leftStyle: indicatorStyle('left'),
+  rightStyle: indicatorStyle('right'),
+
+  _addListeners() {
+    this._scrollElement = this._getScrollElement();
+    this._onScroll = this._updateIndicatorShow.bind(this);
```

Comment:
I think this was intended to be an ember runloop `bind`. Generally we should just use arrow funcs and `run(` imo